task/sp_measure/build.rs

```diff
@@ -0,0 +1,50 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use serde::Deserialize;
+use sha2::{Digest, Sha256};
+use std::io::Write;
+use std::path::PathBuf;
+
+#[derive(Debug, Deserialize)]
+struct TaskConfig {
+    binary_path: PathBuf,
+}
+
+const TEST_SIZE: usize = 0x0010_0000;
+
+fn main() -> Result<(), Box<dyn std::error::Error>> {
+    let out_dir = std::env::var("OUT_DIR")?;
+    let dest_path = std::path::Path::new(&out_dir).join("expected.rs");
+    let mut file = std::fs::File::create(&dest_path)?;
+
+    let task_config = build_util::task_config::<TaskConfig>()?;
+
+    let bin = match std::fs::read(&task_config.binary_path) {
+        Ok(b) => b,
+        Err(_) => vec![0; 256],
```

Comment:
I'd prefer to error out here if the target `.bin` file doesn't exist, to avoid weird confusing situations – is there a reason we want to continue with an invalid checksum?

Also, should we be rerunning if the binary path changes, e.g.
```
    println!("cargo:rerun-if-changed={}", task_config.binary_path);
```
?

Reply:
The problem is that for cases like CI we may not have a binary that exists. I didn't have a better idea of how to merge this code. There's also no guarantee what's flashed on the SP matches anyway. I do think there's value in continuing with an invalid hash to demonstrate how the RoT -> SP SWD control works and for testing anything related to that.

You are correct, I forgot to add the rerun line!